src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -1070,32 +1062,42 @@ public long getCardinalityForField(String fieldName, Date begin, Date end) throw
     public long getCardinalityForField(String fieldName, String datatype, Date begin, Date end) throws TableNotFoundException {
         log.trace("getCardinalityForField from table: " + metadataTableName);
         Text row = new Text(fieldName.toUpperCase());
+        FrequencyFamilyCounter dateFreqMap = new FrequencyFamilyCounter();
         
         // Get all the rows in DatawaveMetadata for the field, only in the 'f'
-        // colfam
+        // colfamily
         Scanner bs = ScannerHelper.createScanner(connector, metadataTableName, auths);
         
         Key startKey = new Key(row);
         bs.setRange(new Range(startKey, startKey.followingKey(PartialKey.ROW)));
         bs.fetchColumnFamily(ColumnFamilyConstants.COLF_F);
         
         long count = 0;
+        Value aggregatedValue = null;
+        
+        String cq = "";
         
         for (Entry<Key,Value> entry : bs) {
-            Text colq = entry.getKey().getColumnQualifier();
+            if (entry.getKey().getColumnQualifier().toString().startsWith(COL_QUAL_PREFIX))
+                aggregatedValue = entry.getValue();
+            cq = entry.getKey().getColumnQualifier().toString().replaceAll(MetadataHelper.COL_QUAL_PREFIX, "");
+        }
+        
+        if (aggregatedValue != null) {
```

Comment:
This method must be backward compatible.  We see need to be able to parse the old style entries.

Reply:
OK. I will get to work on this.  I have automatically assumed that the new iterator will be used since it is added via the MetadataTableConfigHelper class.   There is a test for this backward compatibility in the main project but you are correct in that it will only work if  Datawave is not running the FrequencyTransformIterator.